src/ducks/claimableBalances.ts

```diff
@@ -0,0 +1,73 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import StellarSdk from "stellar-sdk";
+import { RootState } from "config/store";
+import { settingsSelector } from "ducks/settings";
+import { getClaimableBalances } from "helpers/getClaimableBalances";
+import { getErrorString } from "helpers/getErrorString";
+import { getNetworkConfig } from "helpers/getNetworkConfig";
+import {
+  ActionStatus,
+  RejectMessage,
+  ClaimableBalancesInitialState,
+  ClaimableBalance,
+} from "types/types.d";
+
+export const fetchClaimableBalancesAction = createAsyncThunk<
+  {
+    data: ClaimableBalance[];
+  },
+  string,
+  { rejectValue: RejectMessage; state: RootState }
+>(
+  "claimableBalances/fetchClaimableBalancesAction",
+  async (publicKey, { rejectWithValue, getState }) => {
+    const { isTestnet } = settingsSelector(getState());
+    const networkConfig = getNetworkConfig(isTestnet);
+    const server = new StellarSdk.Server(networkConfig.url);
+
+    let data: ClaimableBalance[] = [];
+
+    try {
+      data = await getClaimableBalances({ server, publicKey });
+    } catch (error) {
+      return rejectWithValue({
+        errorString: getErrorString(error),
+      });
+    }
+
+    return {
+      data,
+    };
+  },
+);
+
+const initialClaimableBalancesState: ClaimableBalancesInitialState = {
+  data: [],
+  errorString: undefined,
+  status: undefined,
+};
+
+export const claimableBalancesSlice = createSlice({
+  name: "claimableBalances",
+  initialState: initialClaimableBalancesState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(fetchClaimableBalancesAction.pending, (state) => {
+      state.status = ActionStatus.PENDING;
+      state.errorString = undefined;
+    });
+    builder.addCase(fetchClaimableBalancesAction.fulfilled, (state, action) => {
+      state.data = action.payload.data;
+      state.status = ActionStatus.SUCCESS;
```

Comment:
should `errorString` be set back to `undefined` when the action is fulfilled?

Reply:
I don't think we need to reset `errorString` here because it will be reset in a pending state. There shouldn't be a case where it would skip that and go straight to success. 🤔 Can you think of any scenario where it would happen?

Reply:
Ah, good call!